modules/Microsoft.Sql/servers/.test/secondary/dependencies.bicep

```diff
@@ -0,0 +1,35 @@
+@description('Required. The name of the server.')
+param serverName string
+
+@description('Required. The name of the database.')
+param databaseName string
+
+@description('Optional. The location to deploy resources to.')
+param location string = resourceGroup().location
+
+@description('Optional. The password to leverage for the login.')
+@secure()
+param password string = newGuid()
+
+module serverDeployment '../../deploy.bicep' = {
```

Comment:
@eriqua we did not have the case yet that a dependency would reference itself. For consistency I'm inclined to ask to use a native `resource` here instead. But, it's also a special case, in that the deployment remains in the same module.

Reply:
this could look like:

```Bicep
resource server 'Microsoft.Sql/servers@2021-11-01' = {
  name: serverName
  location: location
  properties: {
    administratorLogin: 'adminUserName'
    administratorLoginPassword: password
  }

  resource database 'databases@2021-11-01' = {
    name: 'db1'
    location: location
    sku: {
      name: 'Basic'
      tier: 'Basic'
    }
    properties: {
      maxSizeBytes: 2147483648
    }
  }
}

@description('The resource ID of the created database.')
output databaseResourceId string = server::database.id
```

Reply:
The upsite is, it would also make the output easier.

Reply:
Changed this to suggestion